### ISSUE TITLE
Hook into JS and CSS

### ISSUE THREAD

**Issue by jenstornell** [distantnative/panel-bar#37]
I'm thinking plugins with new helpers / elements. They are created like described here:

https://github.com/distantnative/panel-bar/issues/28

Probably there is a need for adding some custom CSS and custom JS in there somehow. The plugin maker can ask for adding the extra JS and CSS into the template but to make it less bloated it would be nice to have the panelbar generate that CSS and JS as well.

In conclusion we could just use the panelbar::show() function to have all CSS and JS from the panelbar and from the panelbar plugins.

How? I leave that for now.

Good or bad idea?

**Comment by distantnative** [comment]
I'd solve it by making it possible to pass a parameter `$hook`:

```php
echo panelbar::show(array('js' => false));
…
echo panelbar::css($hook);
```

`$hook` can be a string or a callable function that return a string.

Oh, and I just realized, that ofc it would be nice to be able to pass it to `::show()` as well.

**Comment by distantnative** [comment]
Just doing it with

```php
echo panelbar::show(array('js.hook' => $hook));
```

What do you think?

**Comment by distantnative** [comment]
Implemented on the dev branch.

**Comment by jenstornell** [comment]
I think it looks nice but I have not over-analyzed it yet.

Now I'm just wondering how to do with panel-bar plugins that need CSS or JS.

**Example:**

For some reason someone want to add a Google Map in the panel-bar (don't ask me why). To do that some CSS needs to be added (and possible some JS too).

That plugin would more or less be a custom helper like the loadtime plugin. How should the CSS be inserted? Probably the same way as the user do it. I just need to get my head around it before I can say if there are more stuff to think about here.

**Comment by distantnative** [comment]
I will think about it a lil more, but right now I don't see a good way to get it from such a custom helper into the combined output.

Atm, I'd say the custom element/helper should output the `<div></div>` part followed by a `<style></style>` and in this way include its own CSS (or JS the same way).

**Comment by jenstornell** [comment]
I just tried adding style in the plugin and yes it works. I think it's a totally ok approach. That way the template don't get that messy.

**Comment by distantnative** [comment]
:+1:
